app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaViewModel.kt

```diff
@@ -64,6 +66,16 @@ class AgendaViewModel @Inject constructor(
                 createFakeAgendaItems(agendaRepository)
             }
         }
+
+        // Collect the last state of agendaItems and save it for fast access.
+        // Is there a better way to do this? Better with .stateIn?
+        viewModelScope.launch {
+            agendaState.collect {
+                it.agendaItems.collect {
+                    lastAgendaItems = it
+                }
+            }
+        }
```

Comment:
Not sure if I get you right here, but do you do this so you don't need to load agenda items from the DB again when the user switches dates? Is this really a performance problem the user will notice?

Reply:
Im using it in the `onEvent` handler to check if the `agendaItem` is indeed a `Task` before sending it to the `AgendaRepository` here:

https://github.com/realityexpander/Tasky/blob/050b32373c3fad67a33a88af9ba621b7319ee571/app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaViewModel.kt#L213-L224

Is there a better way to get the current state of the `flow` of `AgendaItems`?

Reply:
Still not sure if I get you right, but can't you just use the local agenda items list in your state to find the item with the ID that was clicked on and check that? Or just pass the task as a parameter to `TaskToggleCompleted`? It's still unclear to me why you need these nested flows

Reply:
OK, I will try to use the local AgendaItems